osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -358,7 +358,8 @@ private void mergeSelection()
         {
             var mergeableObjects = selectedMergeableObjects;
 
-            if (mergeableObjects.Length < 2)
+            if (mergeableObjects.Length < 2 || (mergeableObjects.All(h => h is not Slider)
+                                                && Precision.AlmostBigger(1, Vector2.DistanceSquared(mergeableObjects[0].Position, mergeableObjects[1].Position))))
```

Comment:
Can this be split to a separate method or something, now that there are actual complications? Something like `private bool canMerge(IEnumerable<OsuHitObject> objects)`?

I'm asking because `GetContextMenuItemsForSelection()` does the same check, but negated, _but_ at a glance it doesn't _look_ like the same check (while yes, the check there is correctly negated, de Morgan's laws and all, I feel like that is hard to spot and therefore prone to future error from someone updating one place but not the other).